### DISCUSSION TITLE
OpenTelemetry tracing improvements

### DISCUSSION THREAD

**Discussion by jangaraj** [grafana/grafana#51070] [Observability]
Grafana 9.0.0 added initial support for Opentelemetry (OTEL) tracing. It will be fine if there is:
- Grafana OTEL config option for the service name - now it seems to be hardcoded to `grafana`, so it will be a problem to filter if I have multiple Grafanas with enabled OTEL tracing - it can be solved easily if env variable `OTEL_SERVICE_NAME` is supported
- service version uses Grafana version - I deployed Grafana 9.0.0 and `service.version` reports `3.5.4`
- OTEL config options provides more configurability - it seems to be impossible to use TLS tracing endpoints now, maybe also OTLP HTTP protocol
- optional attributes, which I can configure and which will be added to traces - this works atm with env variable `OTEL_RESOURCE_ATTRIBUTES` e.g. `OTEL_RESOURCE_ATTRIBUTES=cloud.account.id=1234567,cloud.platform=dd,cloud.provider=aws,cloud.region=my-region"`
- documentation which OTEL SDK env variables are supported - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration

**Comment by butlerx** [comment]
Support for pulling in information about the [host](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v1.9.0/resource#WithHost) or [container](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v1.9.0/resource#WithContainer) Is needed as well so as to be able to decern where a trace is coming from

**Comment by hairyhenderson** [comment]
FYI I filed a bug earlier today on one of these items: https://github.com/grafana/grafana/issues/54037. I'm working on it.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
